src/mercury-agent/mercury/agent/procedures/firmware.py

```diff
@@ -0,0 +1,6 @@
+class FirmwareException(Exception):
```

Comment:
I would keep exceptions in their own file and project root or similar, so we don't have an import loop when we want firmware to import hp_firmware, yet hp_firmware needs the exceptions also